src/Client.php

```diff
@@ -79,7 +79,8 @@ public function request($method, $endpoint, $body = null, $headers = [])
             'body' => $body,
             'headers' => array_merge($headers, $defaultHeaders),
         ];
-        if (in_array($method, ['POST', 'PATCH', 'DELETE'])) {
+
+        if (in_array(empty($body) && $method, ['POST', 'PATCH', 'DELETE'])) {
```

Comment:
this `unset`s when empty regardless of method, did we want 
`if (empty($body) && in_array())` instead?